feat: Update MatchTimeline to use v5 struct definition and add EndOfGameResult field to MatchInfo

2ajoyce opened the pull request:
## Update MatchTimeline Structures to v5 & Add EndOfGameResult field to LoL MatchInfo

### Summary
- Updated `MatchTimeline` and related types in model.go to match the documented v5 API structure. Closes #75
- Replaced legacy timeline types with new, more accurate representations (`MetadataTimeline`, `InfoTimeline`, `FramesTimeline`, etc.). 
- Added the missing `EndOfGameResult` property to the League of Legends `MatchInfo` struct to support the complete `/lol/match/v5/matches/{matchId}` API response. Closes #76 

### Changes
- **Added field**: `EndOfGameResult string` with JSON tag `json:"endOfGameResult"`
- **Added tests**: `TestMatchInfo_GetEndOfGameResult` with coverage for multiple scenarios

2ajoyce committed:
patch: Add EndOfGameResult field to MatchInfo and corresponding test

2ajoyce committed:
feat: Update match timeline structures to include v5 fields and add unit tests for new fields

KnutZuidema commented on riot/lol/match.go:
```diff
@@ -131,7 +131,7 @@ func (m *MatchClient) ListStream(puuid string, options ...*MatchListOptions) <-c
 // GetTimeline returns the timeline for the given match
 // NOTE: timelines are not available for every match
-// TODO: update to v5 when struct is documented
+// TODO: double check v5 implementation when struct is documented
```
This TODO can probably be removed now?

2ajoyce replied:
I'm fine either way, I just wasn't sure which you'd prefer. The actual documentation still isn't available, and it looked like you used the api documentation for commented descriptions on many fields.

KnutZuidema replied:
I see. In that case we can leave it in.

KnutZuidema commented on riot/lol/model.go:
```diff
@@ -454,104 +456,152 @@ type Team struct {
-	Timestamp         int                          `json:"timestamp"`
-	ParticipantFrames map[string]*ParticipantFrame `json:"participantFrames"`
-	Events            []*MatchEvent                `json:"events"`
+type MetadataTimeline struct {
```
each of the model types should have at least a short doc comment describing what it represents.

2ajoyce replied:
I left these out because they would be my best guess, not the actual api documentation. I can add them if you'd prefer.

KnutZuidema replied:
Yes, please add these. Your best guess, as the one adding these, is better than no documentation at all.

2ajoyce replied:
Added!

KnutZuidema commented on riot/lol/model_test.go:
```diff
@@ -925,36 +956,224 @@ func TestGameInfo_GetMatch(t *testing.T) {
-			doer: dataDragonResponseDoer(
-				map[string]datadragon.Item{
-					"1": {},
+func TestMatchTimeline_StructFields(t *testing.T) {
```
these tests are useless. There's no need to test that declaration of a struct works.

2ajoyce replied:
I felt the same way writing them, but I saw that some fields had similar tests like `TestParticipant_GetSummoner`, and it did seem like there was some minimal value in having realistic values available in the codebase.
 
Can you clarify the expectations around testing so that we're better aligned next time?

KnutZuidema replied:
Unit tests should test units of logic as defined by the code in the library.

Declaring a struct is not that. In general, a unit test should call some function defined in the library.

2ajoyce replied:
I have removed the tests.

2ajoyce committed:
refactor: remove tests per PR feedback

2ajoyce committed:
refactor: add best guess comments to model.go for match timeline structs

2ajoyce committed:
fix: update GetTimeline method to use region route and add test for region handling

2ajoyce committed:
fix: update GetTimeline test to assert correct API host format for Europe West region

2ajoyce committed:
Merge pull request #1 from 2ajoyce/failing-calls

Update MatchTimeline to use region when constructing url

2ajoyce commented:
@KnutZuidema, I've added an additional change that was necessary to get the method working. It did not have the regionToRoute mapping that the other matchv5 functions were updated to use.

~I've included a test for that, but I know how you felt about the tests I added last time, so let me know if you want me to remove it.~ **Edit:** I was asked to remove this test

KnutZuidema commented on riot/lol/model_test.go:
```diff
@@ -349,6 +349,37 @@ func TestMatchInfo_GetGameMode(t *testing.T) {
 	}
 }
 
+func TestMatchInfo_GetEndOfGameResult(t *testing.T) {
```
this test does not make sense.
There is no `GetEndOfGameResult` method, it just checks that the field is set.
This is not something we need to test, it's a basic Go language feature.

2ajoyce replied:
I have removed this test, per your feedback.

2ajoyce committed:
fix: removed test per repo owner request

KnutZuidema commented:
Sorry, but it looks like we do not understand each other.
I suspect that these changes are primarily made using a coding agent, without really understanding them.
Therefor I am no longer willing to review or merge this PR.
Closing.

2ajoyce commented:
Ok. I'll leave my fork open for anyone trying to use golio. It is non-functional in the current state, but I've tested the fork in a few different implementations and it aligns with the Riot api as of the time of this comment.

Would you like me to close the issues I opened also?
